2013-01-09	Nicolas Setton <setton@adacore.com>

	(gio/gresource.c): Fix loading of gresource files on big-endian
	platforms.
	This patch is already in glib-2.32.2 and above.
	<https://bugzilla.gnome.org/show_bug.cgi?id=673409>

2012-04-02	Nicolas Setton  <setton@adacore.com>

	(glib/gthread-posix.c, glib/tests/thread.c): Some old versions
	of linux have a prctl.h which does not define PR_GET_NAME:
	add test for this.

2012-03-06	Nicolas Setton  <setton@adacore.com>

	(configure): Do not set HAVE_LIBELF, not working on older
	version of Linux (SuSE 10, Red Hat 4)

	Python 2.5 and do not work on old versions of Linux and Solaris.

--- configure.orig	2012-04-18 18:48:43.000000000 +0200
+++ configure	2012-04-18 18:49:26.000000000 +0200
@@ -26060,7 +26060,7 @@
 $as_echo "$ac_cv_lib_elf_elf_begin" >&6; }
 if test "x$ac_cv_lib_elf_elf_begin" = xyes; then :
   cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBELF 1
+#undef HAVE_LIBELF 
 _ACEOF
 
   LIBS="-lelf $LIBS"
@@ -26107,7 +26107,7 @@
 $as_echo "$ac_cv_lib_elf_elf_getshdrstrndx" >&6; }
 if test "x$ac_cv_lib_elf_elf_getshdrstrndx" = xyes; then :
   cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBELF 1
+#undef HAVE_LIBELF
 _ACEOF
 
   LIBS="-lelf $LIBS"
@@ -26154,7 +26154,7 @@
 $as_echo "$ac_cv_lib_elf_elf_getshdrnum" >&6; }
 if test "x$ac_cv_lib_elf_elf_getshdrnum" = xyes; then :
   cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBELF 1
+#undef HAVE_LIBELF
 _ACEOF
 
   LIBS="-lelf $LIBS"
@@ -26180,7 +26180,7 @@
 
 if test $have_libelf = yes; then
 
-$as_echo "#define HAVE_LIBELF 1" >>confdefs.h
+$as_echo "#undef HAVE_LIBELF" >>confdefs.h
 
 fi
 
*** gio/Makefile.in.orig	2012-04-18 18:00:04.000000000 +0200
--- gio/Makefile.in	2012-04-18 18:20:22.000000000 +0200
***************
*** 479,485 ****
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = gdbus-2.0/codegen xdgmime inotify fen win32 . fam tests
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
--- 479,485 ----
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = gdbus-2.0/codegen xdgmime inotify fen win32 . fam
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
***************
*** 756,762 ****
  TEST_PROGS = 
  NULL = 
  SUBDIRS = gdbus-2.0/codegen $(am__append_1) $(am__append_4) \
! 	$(am__append_8) $(am__append_11) . $(am__append_14) tests
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
--- 756,762 ----
  TEST_PROGS = 
  NULL = 
  SUBDIRS = gdbus-2.0/codegen $(am__append_1) $(am__append_4) \
! 	$(am__append_8) $(am__append_11) . $(am__append_14) 
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
--- glib/gthread-posix.c.orig	2012-04-02 10:14:29.000000000 +0200
+++ glib/gthread-posix.c	2012-04-02 10:14:48.000000000 +0200
@@ -1172,8 +1172,10 @@
 g_system_thread_set_name (const gchar *name)
 {
 #ifdef HAVE_SYS_PRCTL_H
+#ifdef PR_SET_NAME
   prctl (PR_SET_NAME, name, 0, 0, 0, 0);
 #endif
+#endif
 }
 
 /* {{{1 Epilogue */
--- glib/tests/thread.c.orig	2012-04-02 10:16:08.000000000 +0200
+++ glib/tests/thread.c	2012-04-02 10:16:29.000000000 +0200
@@ -168,12 +168,14 @@
 thread6_func (gpointer data)
 {
 #ifdef HAVE_SYS_PRCTL_H
+#ifdef PR_GET_NAME
   const gchar name[16];
 
   prctl (PR_GET_NAME, name, 0, 0, 0, 0);
 
   g_assert_cmpstr (name, ==, (gchar*)data);
 #endif
+#endif
 
   return NULL;
 }
--- gio/gresource.c.orig	2013-01-09 11:37:43.136887171 +0100
+++ gio/gresource.c	2013-01-09 11:38:29.297722363 +0100
@@ -287,7 +287,7 @@
       free_path[path_len-1] = 0;
     }
 
-  value = gvdb_table_get_value (resource->table, path);
+  value = gvdb_table_get_raw_value (resource->table, path);
 
   if (value == NULL)
     {
@@ -305,6 +305,9 @@
                      &_flags,
                      &array);
 
+      _size = GUINT_FROM_LE (_size);
+      _flags = GUINT_FROM_LE (_flags);
+
       if (size)
         *size = _size;
       if (flags)
